File: packages/backend/model/geomorph_data.py
from typing import Literal, Optional, TypeAlias
from pydantic import BaseModel


ChannelType: TypeAlias = Literal["fluvial", "transitional", "colluvial"]


class GeomorphData(BaseModel):
    id: str
    stream: int
    ksn: float
    theta_chi: float
    theta_sa: float
    rfit_theta_tt: float
    error_tt: float
    rfit_theta_tak: float
    error_tak: float
    interpretable: bool
    interpret_confidence: float
    interpretable_user: Optional[bool]
    channel_type: Optional[ChannelType]

File: packages/frontend/src/lib/types/GeomorphData.ts
import type { ChannelType } from './ChannelType';

export interface GeomorphData {
	id: string;
	stream: number;
	ksn: number;
	theta_chi: number;
	theta_sa: number;
	rfit_theta_tt: number;
	error_tt: number;
	rfit_theta_tak: number;
	error_tak: number;
	interpretable: boolean;
	interpret_confidence: number;
	interpretable_user?: boolean;
	channel_type?: ChannelType;
}
